File: 2026-08/src/day_29/prms_120894/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

string words[10] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

long long solution(string numbers) {
    for (int d = 0; d < 10; d++) {
        int pos;
        while ((pos = numbers.find(words[d])) != -1) {
            numbers.replace(pos, words[d].size(), to_string(d));
        }
    }

    return stoll(numbers);
}

File: 2026-08/src/day_29/prms_120894/Solution.py
def solution(numbers):
    words = [
        "zero",
        "one",
        "two",
        "three",
        "four",
        "five",
        "six",
        "seven",
        "eight",
        "nine",
    ]

    for d, w in enumerate(words):
        numbers = numbers.replace(w, str(d))

    return int(numbers)
